### ISSUE TITLE
BUG: iirdesign doesn't fail for frequencies above Nyquist

### ISSUE THREAD

**Issue by TomBugnon** [scipy/scipy#11559]
Hi all,

`iirdesign` and the underlying filter order functions (`buttord` and cie) don't seem to check that the `wp` and `wn` input frequencies (normalized by the Nyquist frequency) are within [0, 1], and interpret those normalized frequencies modulo 2.
This might be expected behavior (?) but I think this is extremely error prone, could be made more explicit in the documentation and should at least raise a warning -if not fail- when a sampling rate in provided, since it is quite unlikely anyone is reasoning "modulo 2*fs/2" in this case, isn't it?

```python
import scipy.signal
import numpy as np

fs = 100  # the signal is 100Hz, but maybe I thought it was 1000Hz and didn't check?

# That's fine
wp1, ws1 = np.array([25, 30]), np.array([20, 40])
sos1 = scipy.signal.iirdesign(wp1, ws1, 1, 30, fs=fs, output='sos')

# Now I'm bandpassing between 125,130Hz a signal sampled at 100Hz :(
k = 2
wp2, ws2 = wp1 + k*(fs/2), ws1 + k*(fs/2)
print(wp2, ws2)
sos2 = scipy.signal.iirdesign(wp2, ws2, 1, 30, fs=fs, output='sos')
# No error

# Same results (+- numerical errors?)
print(np.all(sos1-sos2<1e-14))
```

The lack of thorough checking of wp and ws also results in https://github.com/scipy/scipy/issues/11567

#### Scipy/Numpy/Python version information:
1.4.1 1.18.1 sys.version_info(major=3, minor=7, micro=5, releaselevel='final', serial=0)

**Comment by ilayn** [comment]
This is kind of expected since otherwise it is yet another annoyance that a user should slice an array (often with unknown contents say generated via `np.logspace`) until it falls below the nyquist freq. I have the opinion that this should be documented properly but kept as valid input. 

Same applies to Bode plots for example, a user can still pump frequencies beyond the Nyquist but they probably should realize that it is meaningless beyond the dashed line. 

Of course our handling here is still substandard and needs some clarification.

**Comment by TomBugnon** [comment]
@larsoner so what's to be changed in conclusion? wp and ws above [0, 1] should remain valid input but we just add a warning to the doc? Or should we add a kwarg that makes values above 1 fail or raise a warning?

**Comment by larsoner** [comment]
I think we should throw an error. The likelihood that someone intentionally puts a frequency above Nyquist is way lower than them doing it unintentionally, and thus potentially getting an errant result and not realizing it. If anyone really did put a frequency above Nyquist intentionally, it's easy enough for them to alias it back with modulo.

**Comment by TomBugnon** [comment]
I agree. Ok with you @ilayn ?

**Comment by TomBugnon** [comment]
@ilayn I think we need to reopen this.

The check implemented in #13115 by @AtsushiSakai assumes that Wp and Ws are already normalized by the Nyquist frequency, whereas this normalization happens later on (in the `buttord`, `cheb1ord` etc functions). 

As a consequence iirdesign now fails when:
- Wp > 1 or Ws > 1 and sf=None (correct behavior)
but also when:
- Wp > 1, Ws > 1, sf is not None and 0 < (Wp/(sf/2)) , (Ws/(sf/2)) < 1. ( incorrect behavior)
